pkg/cloud/gcp/actuators/machine/machine_scope_test.go

```diff
@@ -0,0 +1,445 @@
+				if err != nil {
+					return fmt.Errorf("unable to get provider status: %v", err)
+				}
+
+				if providerStatus.InstanceID == nil || *providerStatus.InstanceID != "123" {
+					return fmt.Errorf("instanceID is nil or not equal expected \"123\"")
+				}
+
+				if providerStatus.InstanceState == nil || *providerStatus.InstanceState != "running" {
+					return fmt.Errorf("instanceState is nil or not equal expected \"running\"")
+				}
+
+				return nil
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			gs := NewWithT(t)
+			timeout := 10 * time.Second
+
+			// original objects
+			originalProviderSpec := v1beta1.GCPMachineProviderSpec{
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: credentialsSecretName,
+				},
+				UserDataSecret: &corev1.LocalObjectReference{
+					Name: userDataSecretName,
+				},
+			}
+
+			rawProviderSpec, err := v1beta1.RawExtensionFromProviderSpec(&originalProviderSpec)
+			gs.Expect(err).ToNot(HaveOccurred())
+
+			machine := &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      machineName,
+					Namespace: testNamespaceName,
+					Labels:    map[string]string{},
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Machine",
+					APIVersion: "machine.openshift.io/v1beta1",
+				},
+				Spec: machinev1.MachineSpec{
+					ProviderSpec: machinev1.ProviderSpec{
+						Value: rawProviderSpec,
+					},
+				},
+			}
+
+			// Create the machine
+			gs.Expect(k8sClient.Create(ctx, machine)).To(Succeed())
+			defer func() {
+				gs.Expect(k8sClient.Delete(ctx, machine)).To(Succeed())
+			}()
+
+			// Ensure the machine has synced to the cache
+			getMachine := func() error {
+				return k8sClient.Get(ctx, machineKey, machine)
+			}
+			gs.Eventually(getMachine, timeout).Should(Succeed())
+
+			machineScope, err := newMachineScope(machineScopeParams{
+				coreClient:           k8sClient,
+				machine:              machine,
+				Context:              ctx,
+				computeClientBuilder: computeservice.MockBuilderFuncType,
+			})
+
+			gs.Expect(err).ToNot(HaveOccurred())
+
+			tc.mutate(machineScope.machine)
+
+			machineScope.providerStatus = providerStatus
+
+			// Patch the machine and check the expectation from the test case
+			// use Close() instead of Patch(), because Close() sets provider status and spec
+			gs.Expect(machineScope.Close()).To(Succeed())
+			checkExpectation := func() error {
+				if err := getMachine(); err != nil {
+					return err
+				}
+				return tc.expect(machine)
+			}
+			gs.Eventually(checkExpectation, timeout).Should(Succeed())
+
+			// Check that resource version doesn't change if we call patchMachine() again
+			machineResourceVersion := machine.ResourceVersion
+
+			gs.Expect(machineScope.PatchMachine()).To(Succeed())
+			gs.Eventually(getMachine, timeout).Should(Succeed())
+			gs.Expect(machine.ResourceVersion).To(Equal(machineResourceVersion))
```

Comment:
You should replace this with a consistently, we need to make sure that it doesn't change for a period, not that it hasn't changed the first time we call get which is what's currently happening here I think